src/SprykerEco/Zed/PunchoutCatalogs/Communication/Form/PunchoutCatalogConnectionForm.php

```diff
@@ -84,33 +113,107 @@ protected function addBusinessUnitField(FormBuilderInterface $builder, array $op
 
     /**
      * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     * @param array $options
      *
      * @return $this
      */
-    protected function addAddMappingField(FormBuilderInterface $builder)
+    protected function addConnectionFormatSubForms(FormBuilderInterface $builder, array $options)
     {
-        $builder->add(PunchoutCatalogConnectionTransfer::MAPPING, TextareaType::class, [
-            'required' => false,
-        ]);
+        foreach ($options[static::OPTION_CONNECTION_FORMAT_SUB_FORM_TYPES] as $connectionFormat => $connectionFormatSubFormType) {
+            $builder->add($connectionFormat, $connectionFormatSubFormType, [
+                'mapped' => false,
+                'validation_groups' => static::VALIDATION_GROUP_DISABLED,
+                'label' => false,
+            ]);
+        }
+
+        $this->addConnectionFormatDynamicSubFormListener($builder);
 
         return $this;
     }
 
     /**
-     * @param \Symfony\Component\OptionsResolver\OptionsResolver $resolver
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
      *
      * @return void
      */
-    public function configureOptions(OptionsResolver $resolver): void
+    protected function addConnectionFormatDynamicSubFormListener(FormBuilderInterface $builder): void
     {
-        $resolver->setRequired(static::OPTION_BUSINESS_UNIT_CHOICES);
+        $builder->addEventListener(FormEvents::PRE_SET_DATA, function (FormEvent $event) {
+            /**
+             * @var \Generated\Shared\Transfer\PunchoutCatalogConnectionTransfer $punchoutCatalogConnectionTransfer
+             */
+            $punchoutCatalogConnectionTransfer = $event->getData();
+
+            $this->mapSelectedSubFormToMainForm(
+                $event->getForm(),
+                self::OPTION_CONNECTION_FORMAT_SUB_FORM_TYPES,
```

Comment:
`self`? O_o

Reply:
Unfortunately can't use `static` in anonymous functions, but you are correct, this won't go, refactored it